### PR TITLE
add Milankovitch cycles

### LINKED ISSUE #27: Long term variations in time of equinox are wrong
https://gist.github.com/MartinDix/617152231ff4b4d87ac71a2e4e159a5d calculates the time of the vernal equinox using Insolation.jl. Comparing to data from http://cococubed.asu.edu/data/equinox.dat shows an incorrect long term variation.

equinox

However the long term variation in date of perihelion is correct. Calculation with https://gist.github.com/MartinDix/33396874bbddc1f40049e778af68ac58,  data from http://astropixels.com/ephemeris/ephemeris.html).

perihelion

I believe the equinox error is because the distance_declination function uses the anomalistic year (time between passes through perihelion) rather than the tropical year (time between equinoxes). If the function is modified to use a tropical year of 365.2424 days the equinox is calculated correctly, but the time variation of the perihelion is then messed up. 

I confess I'm confused about whether the anomalies in this function refer to equinoxes or perihelion, so at the moment I can't suggest a fix which gets both the equinox and perihelion correct.

### LINKED ISSUE #29: Using Float32 for latitude/longitude gives incorrect zenith angles
Since ClimaAtmos will use Float32 by default, this should be resolved at some point. The current workaround is to convert to Float64 when calling `instantaneous_zenith_angle` and then convert back to `Float32`.

### PR DESCRIPTION
**Pull request by claresinger**
Closes #27 and closes #29 

- Add look up for orbital parameters from Laskar 2004 dataset. 
- Update dealing with dates to make Insolation work with Float32 values. 
- Add tests and documentation for the new Milankovitch cycles. 
- Specifically test that long-term variations in vernal equinox and perihelion date are correct. 
- Change mean anomaly to be calculated relative to epoch not to vernal equinox. 
- Remove Manifest files.

Docs preview here: https://clima.github.io/Insolation.jl/previews/PR34

### CODE REVIEW

**Comment by claresinger** [comment]
bors try

**Comment by claresinger** [comment]
bors try

File: src/ZenithAngleCalc.jl
Comment (szy21) [inline src/ZenithAngleCalc.jl:142]:
```suggestion
    d, δ, _ = distance_declination_hourangle(FT, date, param_set, eot_correction, milankovitch)
```

File: docs/src/InsolationExamples.md
Comment (tapios) [inline docs/src/InsolationExamples.md:72]: It would be nice to add an example showing insolation for a few days at two different locations, e.g., Pasadena (34.1478° N, 118.1445° W), e.g., in January, and Rovaniemi, Finland (66.5039° N, 25.7294° E), e.g., in June. It'll be good check that the diurnal cycle is also correct.

File: docs/src/ZenithAngleEquations.md
Comment (tapios) [inline docs/src/ZenithAngleEquations.md:39]:
```suggestion
The hour angle $\eta$ measures the longitude difference between the subsolar point position at time $t$ and the current location,
```

File: src/Insolation.jl
Comment (tapios) [inline src/Insolation.jl:38]:
```suggestion
Data are read from file and interpolation functions are created in __init__() method.
```

File: src/Insolation.jl
Comment (tapios) [inline src/Insolation.jl:39]:
```suggestion
The functions are stored as global variables that are used inside Insolation.jl.
```

File: src/Insolation.jl
Comment (tapios) [inline src/Insolation.jl:41]: "this paper" seems to be missing a referent.

File: src/InsolationCalc.jl
Comment (tapios) [inline src/InsolationCalc.jl:17]: I think AU here and throughout should be some parameter semimajor_axis_orbit, to be defined in ClimaParameters, so that if we change the mean planet-sun distance, we get the right behavior.

File: src/InsolationCalc.jl
Comment (tapios) [inline src/InsolationCalc.jl:47]: Below here too: semimajor_axis_orbit instead of AU?

File: src/ZenithAngleCalc.jl
Comment (tapios) [inline src/ZenithAngleCalc.jl:26]: It would seem safer to me, and more readable, to define epoch as a date string, then convert to Julian day.

File: src/ZenithAngleCalc.jl
Comment (tapios) [inline src/ZenithAngleCalc.jl:29]: If _epoch were a date string, it could just be converted to Julian day here, in the same way that t_years is calculated.

File: src/ZenithAngleCalc.jl
Comment (tapios) [inline src/ZenithAngleCalc.jl:46]:
```suggestion
    # true anomaly, radians 
```
(remove equation references throughout; they may not be stable)

File: src/ZenithAngleCalc.jl
Comment (tapios) [inline src/ZenithAngleCalc.jl:56]: As above, replace AU by a parameter semimajor_axis_orbit (or something like it).

File: src/ZenithAngleCalc.jl
Comment (tapios) [inline src/ZenithAngleCalc.jl:64]: Why t0 here? It seems confusing because t0 previously referred to time of epoch, but not here (I believe).

File: src/ZenithAngleCalc.jl
Comment (tapios) [inline src/ZenithAngleCalc.jl:105]:
```suggestion
    # zenith angle, radians 
```

File: src/ZenithAngleCalc.jl
Comment (tapios) [inline src/ZenithAngleCalc.jl:122]:
```suggestion
Returns the effective zenith angle corresponding to diurnally averaged insolation and earth-sun distance
```
(because what's averaged is the cosine of the zenith angle)

File: src/ZenithAngleCalc.jl
Comment (tapios) [inline src/ZenithAngleCalc.jl:144]:
```suggestion
    # sunrise/sunset angle 
```

File: src/ZenithAngleCalc.jl
Comment (tapios) [inline src/ZenithAngleCalc.jl:154]:
```suggestion
    # effective zenith angle to get diurnally averaged insolation (i.e., averaging cosine of zenith angle)
```

File: test/test_insolation.jl
Comment (tapios) [inline test/test_insolation.jl:77]: This test was nice. Why not keeping it? (It doesn't seem to depend on the calendar conventions.)

**Review by tapios — APPROVED** [review]
This looks great to me! Thanks so much Clare. Some comments in the file. Also, the documentation landing page still is quite Earth-centric, while you made the rest more planet-agnostic. You might want to do the same on the landing page.

File: test/test_insolation.jl
Comment (claresinger) [reply test/test_insolation.jl:77]: I removed it because I was planning to remove the parameter `lon_perihelion` from CLIMAParameters since it's no longer used. But I can keep the test and just change the `lon_perihelion_epoch` instead. That'll be the same thing.

File: docs/src/ZenithAngleEquations.md
Comment (claresinger) [reply docs/src/ZenithAngleEquations.md:41]: Yes, that is how it's done in the code (wrt to zero longitude and then longitude is added). I updated the docs to reflect this.

File: src/InsolationCalc.jl
Comment (claresinger) [reply src/InsolationCalc.jl:17]: Thanks for catching this. It is already in CLIMAParameters! I just wasn't using it here. `orbit_semimaj` defined as `1*astro_unit()` for `AbstractEarthParameterSet`

File: src/ZenithAngleCalc.jl
Comment (claresinger) [reply src/ZenithAngleCalc.jl:26]: I've updated the code locally. Will need to update CLIMAParameters first before merging.

File: src/ZenithAngleCalc.jl
Comment (claresinger) [inline src/ZenithAngleCalc.jl:29]: Waiting to update CLIMAParameters so that this can be moved from here to there. Will merge when that is ready.

Comment (tapios) [reply src/ZenithAngleCalc.jl:29]: It'll be good to have this string support in ClimaParameters. We'll need more date strings there (e.g., to specify aggregation timeframes). @odunbar

**Comment by claresinger** [comment]
bors try

File: src/ZenithAngleCalc.jl
Comment (claresinger) [reply src/ZenithAngleCalc.jl:29]: @tapios I am going to merge this now and will update Insolation to use the `epoch` value from CLIMAParameters when it is available (i.e., after https://github.com/CliMA/CLIMAParameters.jl/pull/88 is merged).

**Pull request merged**
